client/extensions/woocommerce/state/sites/shipping-zone-locations/selectors.js

```diff
@@ -112,8 +112,8 @@ export const areShippingZonesLocationsValid = ( reduxState, siteId = getSelected
 				}
 				statesSet.add( s );
 			}
-		} else {
-			// A zone must have *any* location. If it doesn't, it's incorrect.
+		} else if ( ! isEmpty( postcode ) ) {
```

Comment:
We must consider a shipping zone with no locations "valid", because it can happen if there is an error saving the zone locations. We shouldn't kick the user to WP-Admin in that situation.